plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Check/Scope.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module UntypedPlutusCore.Check.Scope (checkScope) where
+
+import Control.Lens hiding (index)
+import UntypedPlutusCore.Core.Type as UPLC
+import UntypedPlutusCore.DeBruijn as UPLC
+
+checkScope :: forall name uni fun a. HasIndex name
+           => UPLC.Term name uni fun a
+           -> Bool
```

Comment:
I think it's worth returning a structured error here. Useful for users, easier to test.